File: java/iroha_android/src/main/java/org/hyperledger/iroha/android/offline/attestation/PlayIntegrityTelemetry.java
package org.hyperledger.iroha.android.offline.attestation;

/** Hook invoked whenever a Play Integrity attestation attempt succeeds or fails. */
public interface PlayIntegrityTelemetry {

  PlayIntegrityTelemetry NO_OP =
      new PlayIntegrityTelemetry() {
        @Override
        public void onAttempt(final PlayIntegrityRequest request) {}

        @Override
        public void onSuccess(
            final PlayIntegrityRequest request, final PlayIntegrityAttestation attestation) {}

        @Override
        public void onFailure(final PlayIntegrityRequest request, final Throwable error) {}
      };

  default void onAttempt(final PlayIntegrityRequest request) {}

  void onSuccess(PlayIntegrityRequest request, PlayIntegrityAttestation attestation);

  void onFailure(PlayIntegrityRequest request, Throwable error);
}

File: kotlin/offline-wallet-android/src/main/java/org/hyperledger/iroha/sdk/offline/attestation/PlayIntegrityTelemetry.kt
package org.hyperledger.iroha.sdk.offline.attestation

/** Hook invoked whenever a Play Integrity attestation attempt succeeds or fails. */
interface PlayIntegrityTelemetry {

    fun onAttempt(request: PlayIntegrityRequest) {}

    fun onSuccess(request: PlayIntegrityRequest, attestation: PlayIntegrityAttestation?)

    fun onFailure(request: PlayIntegrityRequest, error: Throwable)

    companion object {
        @JvmField
        val NO_OP: PlayIntegrityTelemetry = object : PlayIntegrityTelemetry {
            override fun onAttempt(request: PlayIntegrityRequest) {}
            override fun onSuccess(request: PlayIntegrityRequest, attestation: PlayIntegrityAttestation?) {}
            override fun onFailure(request: PlayIntegrityRequest, error: Throwable) {}
        }
    }
}
